app/controllers/volunteers_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class VolunteersController < ApplicationController
+    
+before_action :authenticate_user!
+
+  def index
+    # set permission that only the administrator can see the index
+    # also set up function/action either here or in devise that when the user
+        # first sign up for login credentials they're redirected to a creat your
+        # profile page. 
+  # @repair = Repair.all
+    @volunteer = Volunteer.all
+  end
+
+  def new
+  # @repair = Repair.all
+    @volunteer = Volunteer.new
+  end
+  
+  def show
+    @volunteer = current_user.volunteer
+  end
+  
+  def create
+    # @repair = Repair.all
+    @volunteer = Volunteer.new(volunteer_params)
+    @volunteer.user_id = current_user.id
+    @volunteer.save
```

Comment:
There are two @volunteer.save statements.  The first is not needed and should be removed. But I am having trouble understanding what is meant by `redirect_to volunteer`  That does not make sense to me.  You redirect to a path, not to an object.  Tilneil, what does this do when you try it.

Reply:
I made the change. Originally when I was trying to write the code it wouldn't save. Now it does with the @volunteer.save removed. I changed the paths to volunteer_path(@volunteer), or volunteer_path(volunteer). They seem to work and everything seems to be saving. I'm sending another commit.